framework/validators/IpValidator.php

```diff
@@ -0,0 +1,534 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+use yii\helpers\Html;
+use yii\helpers\Json;
+use yii\web\JsExpression;
+
+/**
+ * IpValidator validates that the attribute value is a valid IPv4/IPv6 address or subnet.
+ * May change attribute's value if normalization is enabled.
+ *
+ * @property boolean|string negationChar whether address may have a negation character at the beginning
+ *   string - character passed will be used. May be a regular expression
+ *   true - character from [[DEFAULT_NEGATION_CHAR]] will be used. Default is ```!```
+ *   false  - negation is not allowed
+ * @author SilverFire <d.naumenko.a@gmail.com>
+ * @since 2.0.5
+ */
+class IpValidator extends Validator
+{
+    /**
+     * @const integer the length of IPv6 address in bits
```

Comment:
There's no `@const` in phpdoc. It should be `The length of IPv6 address in bits`.